engine/liveupdate/src/liveupdate_private.cpp

```diff
@@ -54,34 +54,52 @@ namespace dmLiveUpdate
         return NULL;
     }
 
-    uint32_t MissingResources(dmResource::Manifest* manifest, const dmhash_t urlHash, uint8_t* entries[], uint32_t entries_size)
+    uint32_t MissingResources(dmResource::Manifest* manifest, const dmhash_t url_hash, uint8_t* entries[], uint32_t entries_size)
     {
         uint32_t resources = 0;
 
-        if (manifest == 0x0)
+        if (manifest == 0x0 || entries == 0x0 || entries_size == 0)
         {
             return 0;
         }
 
         dmLiveUpdateDDF::HashAlgorithm algorithm = manifest->m_DDFData->m_Header.m_ResourceHashAlgorithm;
         uint32_t hash_len = dmResource::HashLength(algorithm);
 
-        HResourceEntry entry = FindResourceEntry(manifest, urlHash);
+        HResourceEntry entry = FindResourceEntry(manifest, url_hash);
         if (entry != NULL)
         {
             for (uint32_t i = 0; i < entry->m_Dependants.m_Count; ++i)
             {
-                uint8_t* resource_hash = entry->m_Dependants.m_Data[i].m_Data.m_Data;
-                dmResourceArchive::Result result = dmResourceArchive::FindEntry(manifest->m_ArchiveIndex, resource_hash, hash_len, 0, 0);
-                if (result != dmResourceArchive::RESULT_OK)
+                dmhash_t dependant_url_hash = entry->m_Dependants.m_Data[i];
+                HResourceEntry dependant = FindResourceEntry(manifest, dependant_url_hash);
```

Comment:
use the shorter (8 byte) url_hash, vs the cryptographic hash (40 bytes for sha256) to find the resource entry.